lib/internal/assert/calltracker.js

```diff
@@ -36,7 +38,8 @@ class CallTrackerContext {
   }
 
   track(thisArg, args) {
-    ArrayPrototypePush(this.#calls, { thisArg, arguments: args });
+    args = ObjectFreeze(ArrayPrototypeSlice(args));
```

Comment:
i think it's still a deopt to assign to an argument - maybe make a new variable here?
```suggestion
    const argsCopy = ObjectFreeze(ArrayPrototypeSlice(args));
```